jena-fuseki2/jena-fuseki-webapp/src/main/java/org/apache/jena/fuseki/mgt/ActionDatasets.java

```diff
@@ -151,14 +151,16 @@ else if ( WebContent.isMultiPartForm(ct) )
             else
                 assemblerFromBody(action, dest);
 
+            Model model = ModelFactory.createDefaultModel();
+            model.add(modelData);
```

Comment:
Looks like there will be two `Model` instances here, but `model` is never used, except for creating a copy of `modelData` to be used in `AssemblerUtils.addRegistered`. The rest of the code uses `modelData`.

If that's correct, maybe that could be `AssemblerUtils.addRegistered(ModelFactory.createDefaultModel().add(modelData));` to avoid risking the `model` instance being used accidentally. Or maybe `Model` could have a `.copy()` method?

Reply:
It does get used but its via the actions of the Model API and `FusekiConfig.buildDataAccessPoint`.

`model` is used first by `findService` and then `FusekiConfig.buildDataAccessPoint(subject, registry)` which walks the model. A `Resource` knows which model it came from. (e.g. `subject.listProperties` needs the model of the subject).

`AssemblerUtils.addRegistered(model)` modifies the model argument. We want to write `modelData` but after the assembling is done and so is known to be successful (there is also a model write at L162, `systemFileCopy` which is really logging.) Being after, we need the unmodified and modified versions around.

`model.add(anOtherModel)` is not that expensive - it shares strings used in URIs, strings in literals and blank nodes.
(Adding one graph into another is cheaper.)